lib/icinga/checkable-dependency.cpp

```diff
@@ -145,32 +297,94 @@ std::set<Checkable::Ptr> Checkable::GetChildren() const
 	return parents;
 }
 
+/**
+ * Retrieve the total number of all the children of the current Checkable.
+ *
+ * Note, due to the max recursion limit of 256, the returned number may not reflect
+ * the actual total number of children involved in the dependency chain.
+ *
+ * @return int - Returns the total number of all the children of the current Checkable.
+ */
+size_t Checkable::GetAllChildrenCount() const
+{
+	// Are you thinking in making this more efficient? Please, don't.
+	// In order not to count the same child multiple times, we need to maintain a separate set of visited children,
+	// which is basically the same as what GetAllChildren() does. So, we're using it here!
+	return GetAllChildren().size();
+}
+
 std::set<Checkable::Ptr> Checkable::GetAllChildren() const
 {
-	std::set<Checkable::Ptr> children = GetChildren();
+	std::set<Checkable::Ptr> children;
 
 	GetAllChildrenInternal(children, 0);
 
 	return children;
 }
 
-void Checkable::GetAllChildrenInternal(std::set<Checkable::Ptr>& children, int level) const
+/**
+ * Retrieve all direct and indirect children of the current Checkable.
+ *
+ * Note, this function performs a recursive call chain traversing all the children of the current Checkable
+ * up to a certain limit (256). When that limit is reached, it will log a warning message and abort the operation.
+ *
+ * @param seenChildren - A container to store all the traversed children into.
+ * @param level - The current level of recursion.
+ */
+void Checkable::GetAllChildrenInternal(std::set<Checkable::Ptr>& seenChildren, int level) const
 {
-	if (level > 32)
+	if (level > l_MaxDependencyRecursionLevel) {
+		Log(LogWarning, "Checkable")
+			<< "Too many nested dependencies (>" << l_MaxDependencyRecursionLevel << ") for checkable '" << GetName() << "': aborting traversal.";
 		return;
+	}
 
-	std::set<Checkable::Ptr> localChildren;
+	for (const Checkable::Ptr& checkable : GetChildren()) {
+		if (seenChildren.find(checkable) == seenChildren.end()) {
+			seenChildren.emplace(checkable);
```

Comment:
`insert()` already returns whether it actually inserted a new value, so both of this can be done with a single call.